vhdbuilder/packer/imagecustomizer/scripts/publish-imagecustomizer-image.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+set -e
+
+source ./parts/linux/cloud-init/artifacts/cse_benchmark_functions.sh
+
+# Find the absolute path of the directory containing this script
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+CONFIG=$IMG_CUSTOMIZER_CONFIG
+AGENTBAKER_DIR=`realpath $SCRIPTS_DIR/../../../../`
+OUT_DIR="${AGENTBAKER_DIR}/out"
+CREATE_TIME="$(date +%s)"
+
+required_env_vars=(
+    "AZURE_MSI_RESOURCE_STRING"
+    "RESOURCE_GROUP_NAME"
+    "SIG_IMAGE_NAME"
+    "SUBSCRIPTION_ID"
+    "CAPTURED_SIG_VERSION"
+    "PACKER_BUILD_LOCATION"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+# Default to this hard-coded value for Linux does not pass this environment variable into here
+if [ -z "$SIG_GALLERY_NAME" ]; then
+  SIG_GALLERY_NAME="PackerSigGalleryEastUS"
+fi
+
+capture_benchmark "${SCRIPT_NAME}_prepare_upload_vhd_to_blob"
+
+echo "Uploading ${OUT_DIR}/${CONFIG}.vhd to ${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd"
+
+echo "Setting azcopy environment variables with pool identity: $AZURE_MSI_RESOURCE_STRING"
+export AZCOPY_AUTO_LOGIN_TYPE="MSI"
+export AZCOPY_MSI_RESOURCE_STRING="$AZURE_MSI_RESOURCE_STRING"
+export AZCOPY_CONCURRENCY_VALUE="AUTO"
+azcopy copy "${OUT_DIR}/${CONFIG}.vhd" "${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd" --recursive=true || exit $?
```

Comment:
we should cleanup/remove `"${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd"` if we're not planning on generating publishing info (such as during an official release), otherwise just having the gallery image should be sufficient - this is what we do for all over distros

Reply:
Added a check on GENERATE_PUBLISHING_INFO and we'll now clean up the storage account VHD if we aren't using it for publishing later. Confirmed both cases work in the test pipeline as well.